lib/vector-buffers/src/lib.rs

```diff
@@ -84,6 +84,18 @@ pub trait EventCount {
     fn event_count(&self) -> usize;
 }
 
+impl<T> EventCount for Vec<T> {
+    fn event_count(&self) -> usize {
+        self.len()
+    }
+}
+
+impl<T> EventCount for &Vec<T> {
+    fn event_count(&self) -> usize {
+        self.len()
+    }
+}
```

Comment:
Could this be used with `Vec<EventArray>` to do the wrong thing?

Reply:
It could, yes. I'll file an issue to re-examine how I approached wiring up `EventCount` to accomplish my work in `RequestMetadata.

I was really struggling to find the right fit in terms of getting the correct count of events for arbitrary event inputs to `RequestBuilder`. In particular, I did this because the `loki` sink -- maybe others? I was just changing/adding trait impls until `cargo check` was happy -- uses `Vec<LokiRecord>` for its input events to the request builder.

In hindsight, the simpler solution was probably to bound the `EventCount` impl for `T: EventCount` and do a sum, and then impl `EventCount` for `LokiRecord`. I'll probably anchor the issue around taking that approach, unless other usages require a different/cleaner approach.